### ISSUE TITLE
Building on Mac OS 10.14 failure - Linking Premake4 ld: library not found for -lgcc_s.10.4

### ISSUE THREAD

**Issue by AllenDang** [wxFormBuilder/wxFormBuilder#525]
Invoke ./create_build_files4.sh failed. I tried to run 

`sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target /`

Doesn't work.

I tried to set `export MACOSX_DEPLOYMENT_TARGET=10.14`  still doesn't work.

**Comment by hschneider66** [comment]
Same problem here. Can someone provide a working macOS binary of the latest build, please?

**Comment by Sisyphus250** [comment]
I have this problem as well, it isn't installing correctly
 create_build_files4.sh references files that are not on my mac thats why I get an error

**Comment by hschneider66** [comment]
it does not build under macOS and not under Windows. The only chance you have is to use outdated binaries. It makes me sad, because the only useable GUI builder seems to disappear slowly ... I ask myself if developing under wx is the right path, if the future looks like this.

**Comment by sodevel** [comment]
> it does not build under macOS and not under Windows

This is not quite correct, it does build on all platforms used by the CI, but depending on platform and the used build system some extra steps are necessary to actually run the build. Apparently these are not fully up to date. Granted, the situation on MacOS is a bit worse, but this platform is really difficult to maintain for the active developers because no one is using this platform or has direct access to it.

> because the only useable GUI builder seems to disappear slowly

Thanks for these warm words. I don't see the project to disappear, i can only speak for myself here but in the company i work we are using wxFB for all our projects and if there is demand i make additions and corrections. But i can devote only limited time for this and currently for us it does work well enough. Sure there are some outstanding issues but fixing some of these is a huge task which i just can't do right now.

This open source project isn't backed by some big entity and therefor depended on the community users to contribute and right now there is some lack of these. So contributions are welcome, i try to process PR's in a timely manner, but much more than that i can't contribute myself currently.

**Comment by hschneider66** [comment]
Thanks for clarifying the situation. Good to know, that it is still active :-)

**Comment by LoganDark** [comment]
I couldn't find any `gcc_s` anywhere using grep, so changing which library the build process looks for is out of the question for me. It's probably automatic anyway, so it might not even be wxFormBuilder's fault.

However, there is a way to make it find what it wants, even if it's looking for `libgcc_s.10.4` - perhaps you have `libgcc_s.1`? See where I'm going with this?

My solution:
1. Find `libgcc_s.1.dylib` (probably somewhere like `/usr/local/Cellar/gcc/9.3.0_1/lib/gcc/9/libgcc_s.1.dylib` - `brew install gcc && find /usr/local/Cellar/gcc/ -iname 'libgcc_s*.dylib'`)
2. Make a link to it named `libgcc_s.10.4.dylib` (I used `cp` at first, but `ln -s` is probably better):

   ```
   ln -s /usr/local/Cellar/gcc/9.3.0_1/lib/gcc/9/libgcc_s.1.dylib /usr/local/Cellar/gcc/9.3.0_1/lib/gcc/9/libgcc_s.10.4.dylib
   ```
3. Set `LIBRARY_PATH` to the directory containing your newly created link, like so, and run `create_build_files4.sh` like that:

   ```
   LIBRARY_PATH=/usr/local/Cellar/gcc/9.3.0_1/lib/gcc/9/ ./create_build_files4.sh
   ```
4. The script should succeed, and the rest of the build steps should work mostly fine. When running `make`, though, the linker will give you an interactive prompt asking for the location of `libwx_osx_cocoau_stc-3.1.dylib` or similar. For me it accepted `/usr/local/lib`, and it'll probably be the same for you too.

   ```
   * Collecting dependencies...
   /!\ WARNING : Library libwx_osx_cocoau_stc-3.1.dylib  has an incomplete name (location unknown)
   Please specify now where this library can be found (or write 'quit' to abort): /usr/local/lib
   
   /usr/local/lib/libwx_osx_cocoau_stc-3.1.dylib  was found.
   ```

Here's a binary built with the **latest macOS Mojave** using this method. It's dynamically linked against wxWidgets 3.1.3, so if you don't have that, just build it from source (as a dynamic library) and `make install` it.

[wxFormBuilder.app.zip](https://github.com/wxFormBuilder/wxFormBuilder/files/4941230/wxFormBuilder.app.zip)

Here's what it looks like in dark mode:

![image](https://user-images.githubusercontent.com/4723091/87846861-fbde2000-c887-11ea-840e-6813d51a7cce.png)

The wxFormBuilder GUI itself is a bit hideous, but the actual content (a tiny login form I whipped up for the screenshot) looks fine, which is all that matters in my opinion.
